python/lsst/ip/isr/linearize.py

```diff
@@ -679,6 +680,52 @@ def __call__(self, image, **kwargs):
             return False, 0
 
 
+class LinearizeSpline(LinearizeBase):
+    """Correct non-linearity with a spline model.
+
+    corrImage = uncorrImage - Spline(coeffs, uncorrImage)
+
+    Notes
+    -----
+
+    The spline fit calculates a correction as a function of the
+    expected linear flux term.  Because of this, the correction needs
+    to be subtracted from the observed flux.
+
+    """
+    LinearityType = "Spline"
+
+    def __call__(self, image, **kwargs):
+        """Correct for non-linearity.
+
+        Parameters
+        ----------
+        image : `lsst.afw.image.Image`
+            Image to be corrected
+        kwargs : `dict`
+            Dictionary of parameter keywords:
+            ``"coeffs"``
+                Coefficient vector (`list` or `numpy.array`).
+            ``"log"``
+                Logger to handle messages (`lsst.log.Log`).
```

Comment:
I'm actually not familiar with documenting kwargs - does this mean it's considered a mandatory kwarg? Either way, it's not made use of here.

Reply:
The lookup table linearizer has log output, and with the goal of "the linearizers should quack identically", I've taken that to mean that they should accept a log kwarg, even if it's unused.